Source/DafnyLanguageServer/Language/DiagnosticErrorReporter.cs

```diff
@@ -14,8 +15,7 @@ public class DiagnosticErrorReporter : ErrorReporter {
     private readonly DocumentUri entryDocumentUri;
     private readonly Dictionary<DocumentUri, List<Diagnostic>> diagnostics = new();
     private readonly Dictionary<DiagnosticSeverity, int> counts = new();
-
-    public IReadOnlyDictionary<DocumentUri, List<Diagnostic>> Diagnostics => diagnostics;
+    private System.Threading.ReaderWriterLockSlim rwLock = new();
```

Comment:
Mark as `readonly`. And you may add `using System.Threading` in-place of the fully qualified name.

Reply:
Done